### PR TITLE
Adding i18n to records

### PR DESCRIPTION
**Pull request by f-necas**
This PR is intended to get user current locale in records instead of the default one.

If a translation is available in the record for his language, it uses this one.

Issue #556

### CODE REVIEW

File: libs/feature/search/src/lib/utils/mapper/elasticsearch.field.mapper.ts [hunk 1]
```diff
@@ -31,9 +33,14 @@ type EsFieldMapperFn = (
 export class ElasticsearchFieldMapper {
   constructor(
     private metadataUrlService: MetadataUrlService,
-    private linkClassifier: LinkClassifierService
+    private linkClassifier: LinkClassifierService,
+    private translateService: TranslateService
   ) {}
 
+  private readonly lang3 = LANG_2_TO_3_MAPPER[this.translateService.currentLang]
```
Comment (fgravin) [inline libs/feature/search/src/lib/utils/mapper/elasticsearch.field.mapper.ts:40]: Is it a fixed variable ? With the language switcher, I expect people to dynamically change the language of the UI.

File: libs/util/shared/src/lib/fixtures/elasticsearch/full-response.ts [hunk 2]
```diff
@@ -1269,6 +1269,8 @@ export const ES_FIXTURE_FULL_RESPONSE = deepFreeze({
             {
               default: 'Restriction lié à l’exercice du droit moral',
               langfre: 'Restriction lié à l’exercice du droit moral',
+              langger:
+                'Einschränkung im Zusammenhang mit der Ausübung moralischer Rechte',
```
Comment (fgravin) [inline libs/util/shared/src/lib/fixtures/elasticsearch/full-response.ts:1273]: :astonished:

**Review by fgravin — COMMENTED** [review]
Very nice start !
We have to be careful about what fields are multilingual and what are not.
Also, it depends on the versions of GN... for instance, `organizations` & `contacts` have been converted to multilingual very shortly.
It might miss the `contacts` for the last version of GN `4.2.5`.

Also, the hard work, which could be led in another PR, is about the way we run the search.
The index is multilingual, there is a setting to set the language of the catalog (which is more the language of the records of the catalog though). If set to `fre` the search is led on `any.fre` instead of `any.default` (`resourceTitleObject.fre` and all other multilingual fields). You can also have `any.*` etc...

We might need to work with Christophe on this cause he studied that for geocat and see how it's handle in geonetwork vanilla as well. It's not an easy task.
:raised_hands:

File: libs/util/shared/src/lib/fixtures/elasticsearch/full-response.ts (hunk 2 above)
Comment (f-necas) [reply libs/util/shared/src/lib/fixtures/elasticsearch/full-response.ts:1273]: I added a test 😄

File: libs/feature/search/src/lib/utils/mapper/elasticsearch.field.mapper.ts (hunk 1 above)
Comment (f-necas) [reply libs/feature/search/src/lib/utils/mapper/elasticsearch.field.mapper.ts:40]: I don't know what's the actual behavior  when we change locale.Is it reloading the page ? in this case it should work. 

If it doesn't reload, I'll get a bigger look on onLangChange to edit this variable.

File: libs/api/metadata-converter/src/lib/gn4/atomic-operations.ts [hunk 3]
```diff
@@ -22,13 +22,15 @@ export const selectFallback = <T, U>(field: T, fallback: U): T | U =>
   field === null ? fallback : field
 
 export const selectTranslatedValue = <T>(
-  source: SourceWithUnknownProps
-): T | null => selectField(source, 'default')
+  source: SourceWithUnknownProps,
+  lang3: string
+): T | null => selectField(source, lang3) || selectField(source, 'default')
```
Comment (jahow) [inline libs/api/metadata-converter/src/lib/gn4/atomic-operations.ts:27]:
```suggestion
): T | null => selectFallback(selectField(source, lang3), selectField(source, 'default'))
```

**Review by jahow — APPROVED** [review]
Great work!!

**Comment by f-necas** [comment]
Thanks for your reviews ! 
I've updated the fallback thing and also updated code to get translation of contacts and orgs.

**Comment by f-necas** [comment]
Thanks for your reviews ! 
I've updated the fallback thing and also updated code to get translation of contacts and orgs.

**Review by jahow — APPROVED** [review]
Thanks!

**Pull request merged**